applications/fishing-map/src/features/report/report.slice.tsx

```diff
@@ -0,0 +1,159 @@
+
+export const createSingleReportThunk = createAsyncThunk<
+  Report,
+  CreateReport,
+  {
+    rejectValue: AsyncError
+  }
+>('report/createsingle', async (createReport: CreateReport, { rejectWithValue }) => {
+  try {
+    const { dateRange, filters, datasets } = createReport
+    const fromDate = DateTime.fromISO(dateRange.start).toUTC().toISODate()
+    const toDate = DateTime.fromISO(dateRange.end).toUTC().toISODate()
+
+    const queryFiltersFields = [
+      {
+        value: filters.flags,
+        field: 'flag',
+        operator: 'IN',
+        transformation: (value: any): string =>
+          `(${(value as MultiSelectOption<string>[])?.map((f) => `'${f.id}'`).join(', ')})`,
+      },
+      {
+        value: filters.fleets,
+        field: 'fleet',
+        operator: 'IN',
+        transformation: (value: any): string =>
+          `(${(value as MultiSelectOption<string>[])?.map((f) => `'${f.id}'`).join(', ')})`,
+      },
+      {
+        value: filters.origins,
+        field: 'origin',
+        operator: 'IN',
+        transformation: (value: any): string =>
+          `(${(value as MultiSelectOption<string>[])?.map((f) => `'${f.id}'`).join(', ')})`,
+      },
+      {
+        value: filters.geartype,
+        field: 'geartype',
+        operator: 'IN',
+        transformation: (value: any): string =>
+          `(${(value as MultiSelectOption<string>[])?.map((f) => `'${f.id}'`).join(', ')})`,
+      },
+      {
+        value: filters.vessel_type,
+        field: 'vessel_type',
+        operator: 'IN',
+        transformation: (value: any): string =>
+          `(${(value as MultiSelectOption<string>[])?.map((f) => `'${f.id}'`).join(', ')})`,
+      },
+    ]
+
+    const queryFilters = queryFiltersFields
+      .filter(({ value }) => value !== undefined)
+      .map(
+        ({ field, operator, transformation, value }) =>
+          `${field} ${operator} ${transformation ? transformation(value) : `'${value}'`}`
+      )
+
+    const payload = {
+      ...createReport,
+      type: 'detail',
+      timeGroup: 'none',
+      filters: queryFilters,
+      datasets: datasets,
+      dateRange: [fromDate, toDate],
+    }
+    const createdReport = await GFWAPI.fetch<Report>('/v1/reports', {
+      method: 'POST',
+      body: payload as any,
+    })
+    return createdReport
+  } catch (e) {
+    return rejectWithValue({ status: e.status || e.code, message: e.message })
+  }
+})
+
+export const createReportThunk = createAsyncThunk(
+  'report/create',
+  async (createReport: CreateReport[], { dispatch }) => {
+    return Promise.all(createReport.map((report) => dispatch(createSingleReportThunk(report))))
+  }
+)
+
+export interface ReportState extends AsyncReducer<Report> {
+  dialog: {
+    isOpen: boolean
+    geometry: GeoJSON.FeatureCollection | undefined
+    areaName: string
+  }
+}
+
+const initialState: ReportState = {
+  ...asyncInitialState,
+  dialog: {
+    isOpen: false,
+    geometry: undefined,
+    areaName: '',
```

Comment:
Thinking on the report available by url.. I realised we have an id of each feature generated by us called `gfw_id` so I might include it later on so to ensure we can find it in the sources using the `https://docs.mapbox.com/mapbox-gl-js/api/map/#map#querysourcefeatures` instead of the clicked point

Reply:
Makes sense, let's do that.

Reply:
I already started in my branch, will share the definitive code